File: test/easy/IsAnagramTest.php

<?php

use Src\Easy\IsAnagram;

describe('\n\tFeature: Anagram detection', function () {

    test('Testing', function (string $s, string $t, bool $expected) {
        $instance = new IsAnagram;

        expect($instance->solve($s, $t))->toBe($expected);

    })->with([
        ['listen', 'silent', true],
        ['evil', 'vile', true],
        ['race', 'care', true],
        ['anagram', 'nagaram', true],
        ['listen', 'silent', true],
        ['triangle', 'integral', true],
        ['apple', 'pale', false],
        ['aacc', 'ccac', false],
        ['', '', true],
        ['a', 'a', true],
        ['a', 'b', false],
        ['aaaaa', 'aaaaa', true],
        ['abcd', 'abdc', true],
    ]);
});

File: test/easyswift/IsAnagramTest.swift
import Testing

@testable import easy

@Test(arguments: [
  ("anagram", "nagaram", true), ("listen", "silent", true), ("triangle", "integral", true),
  ("apple", "pale", false), ("aacc", "ccac", false),
  // Edge cases
  ("", "", true),  // Two empty strings are anagrams
  ("a", "a", true),  // Single character strings are anagrams
  ("a", "b", false),  // Different single characters are not anagrams
  ("aaaaa", "aaaaa", true),  // Identical strings
  ("abcd", "abdc", true),  // String with repeated characters

]) func isAnagramTest(first: String, second: String, outcome: Bool) {
  #expect(isAnagram(first, second) == outcome)
}

